### PR TITLE
feat: add permission service

### PR DESCRIPTION
**Pull request by muhammedaltug**
- replace config state's getGrantedPolicy selector with permission service methods
- config state's getGrantedPolicy selector deprecated
- add injector to AbstractNavTreeService constructor
- tests fix
- parameter names change

### CODE REVIEW

File: npm/ng-packs/packages/core/src/lib/guards/permission.guard.ts [hunk 1]
```diff
@@ -5,26 +5,31 @@ import { Observable, of } from 'rxjs';
 import { tap } from 'rxjs/operators';
 import { RestOccurError } from '../actions/rest.actions';
 import { RoutesService } from '../services/routes.service';
-import { ConfigState } from '../states/config.state';
 import { findRoute, getRoutePath } from '../utils/route-utils';
+import { PermissionService } from '../services';
```
Comment (bnymncoskuner) [inline npm/ng-packs/packages/core/src/lib/guards/permission.guard.ts]:
```suggestion
import { PermissionService } from '../services/permission.service';
```

File: npm/ng-packs/packages/core/src/lib/tests/dynamic-layout.component.spec.ts [hunk 2]
```diff
@@ -1,18 +1,19 @@
 import { HttpClient } from '@angular/common/http';
-import { Component, NgModule } from '@angular/core';
+import { Component, Injector, NgModule, ɵangular_packages_core_core_d } from '@angular/core';
```
Comment (bnymncoskuner) [inline npm/ng-packs/packages/core/src/lib/tests/dynamic-layout.component.spec.ts]:
```suggestion
import { Component, Injector, NgModule } from '@angular/core';
```

File: npm/ng-packs/packages/core/src/lib/services/permission.service.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { ConfigState } from '../states';
+import { Store } from '@ngxs/store';
+import { map } from 'rxjs/operators';
+import { ApplicationConfiguration } from '../models';
```
Comment (bnymncoskuner) [inline npm/ng-packs/packages/core/src/lib/services/permission.service.ts]:
```suggestion
import { ApplicationConfiguration } from '../models/application-configuration';
```

**Review by bnymncoskuner — APPROVED** [review]
LGTM

**Comment by armanozak** [comment]
Fixes #4588

**Pull request merged**